### ISSUE TITLE
Exporting CIText to Excel throws w/solution

### ISSUE THREAD

**Issue by theronic** [metabase/metabase#5421]
Metabase v0.24.2 throws when trying to export a PostgresSQL column of type CIText to Excel (.xlsx). I ran into exactly this issue recently on a Clojure project using of mine and recognized it immediately. You can overcome the error by casting columns to text, or you can extend the `ISQLResultSetReadColumn` protocol. Solution below.

## Problem:

```
{
  "message": "No method in multimethod 'set-cell!' for dispatch value: class org.postgresql.util.PGobject",
  "stacktrace": ["api.dataset$export_to_xlsx.invokeStatic(dataset.clj:58)", "api.dataset$export_to_xlsx.invoke(dataset.clj:57)", "api.dataset$as_format.invokeStatic(dataset.clj:102)", "api.dataset$as_format.invoke(dataset.clj:94)", "api.dataset$fn__28513$fn__28514.invoke(dataset.clj:123)", "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:229)", "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:224)", "api.dataset$fn__28513.invokeStatic(dataset.clj:116)", "api.dataset$fn__28513.invoke(dataset.clj:116)", "middleware$enforce_authentication$fn__38784.invoke(middleware.clj:120)", "api.routes$fn__38910.invokeStatic(routes.clj:59)", "api.routes$fn__38910.invoke(routes.clj:59)", "routes$fn__39540$fn__39541.doInvoke(routes.clj:64)", "routes$fn__39540.invokeStatic(routes.clj:60)", "routes$fn__39540.invoke(routes.clj:60)", "middleware$log_api_call$fn__38883$fn__38885.invoke(middleware.clj:329)", "middleware$log_api_call$fn__38883.invoke(middleware.clj:328)", "middleware$add_security_headers$fn__38833.invoke(middleware.clj:243)", "middleware$bind_current_user$fn__38788.invoke(middleware.clj:140)", "middleware$maybe_set_site_url$fn__38837.invoke(middleware.clj:266)"]
}
```

## Solution:

Assuming Metabase is using JDBC (I was using funcool's JDBC's core), you can extend `ISQLResultSetReadColumn`, probably here in the [code](https://github.com/metabase/metabase/blob/ce38a6dffc84fd7d72a2ee5e24f6e8937a83aa45/src/metabase/driver/generic_sql.clj#L133).

```
(ns your-namespace.data
  (:import java.sql.SQLException)
  (:require
    [jdbc.core :as jdbc]
    [honeysql.core :as sql]
    [jdbc.transaction :as tx]
    [jdbc.core :as jdbc]
    [jdbc.proto :as proto]))

;; Extend to support citext type:

(import 'org.postgresql.util.PGobject)
(extend-protocol proto/ISQLResultSetReadColumn
  PGobject
  (from-sql-type [pgobj conn metadata index]
    (let [type  (.getType pgobj)
          value (.getValue pgobj)]
      (case type
        "citext" (.toString value)  ;; the magic
        :else value)))
```

## Steps to reproduce:

```
create extension citext;
create table tbl (name citext);
insert into tbl (name) values ('cAse-InsensitiVe text');
select name from tbl; -- exporting this result to XLSX will throw.
select name::text from tbl; -- will not throw
```

**Comment by camsaul** [comment]
Hi @theronic , if you have a sense of what needs to be done we'd love to have you as a contributor and would welcome a PR for it 👍

**Comment by flamber** [comment]
I cannot reproduce on 0.35.3 - think it was fixed since 0.27
